### PR TITLE
Change validation test texture_format_equality to texture_format_compatibility and support copy between compatible formats

### PR DESCRIPTION
**Pull request by shaoboyan**
This PR update api,validation,encoding,cmds,copyTextureToTexture:texture_format_equality
to api,validation,encoding,cmds,copyTextureToTexture:texture_format_compatibility and expand the implementation
to cover copy between compatible format textures.

Issue: #935

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts [hunk 1]
```diff
@@ -265,7 +265,7 @@ g.test('multisampled_copy_restrictions')
     );
   });
 
-g.test('texture_format_equality')
+g.test('texture_format_equality_or_compatibility')
```
Comment (Kangz) [inline src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts:268]: nit: just `texture_format_compatiblity`?

**Review by Kangz — APPROVED** [review]

File: src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts (hunk 1 above)
Comment (shaoboyan) [reply src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts]: Done.

**Comment by kainino0x** [comment]
I couldn't review this before vacation but it looks like it has a review already? Looks like it's really just a change to the description? but the PR title doesn't seem to reflect the change. Please update the title before landing (and make sure it's updated in the "Squash and merge" commit).

**Review by Kangz — APPROVED** [review]
LGTM with @kainino0x's comment.

**Review by Kangz — APPROVED** [review]

**Review by kainino0x — APPROVED** [review]

**Pull request merged**